java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TestClassGenerator.java

```diff
@@ -0,0 +1,324 @@
+			Exceptions.printStackTrace(ex);
+		    }
+		}
+	    }
+	    if (!framework2Add.isEmpty()) {
+		validCombinations.put(location, framework2Add);
+	    }
+	}
+	return validCombinations;
+    }
+
+    private static String getTestMethodName(String methodName) {
+	return "test" + capitalizeFirstLetter(methodName);
+    }
+
+    private static String capitalizeFirstLetter(String str) {
+        if (str == null || str.length() <= 0) {
+            return str;
+        }
+
+        char chars[] = str.toCharArray();
+        chars[0] = Character.toUpperCase(chars[0]);
+        return new String(chars);
+    }
+
+    private static List<String> getTestingFrameworks(FileObject fileObject) {
+	List<String> testingFrameworks = new ArrayList<>();
+	Collection<? extends Lookup.Item<TestCreatorProvider>> testCreatorProviders = Lookup.getDefault().lookupResult(TestCreatorProvider.class).allItems();
+	for (Lookup.Item<TestCreatorProvider> provider : testCreatorProviders) {
+            if (provider.getInstance().enable(new FileObject[]{fileObject})) {
+                testingFrameworks.add(provider.getDisplayName());
+            }
+	}
+        return testingFrameworks;
+    }
+
+    private static Object[] getLocations(FileObject activeFO) {
+        Object[] locations = null;
+	Collection<? extends CommonTestUtilProvider> testUtilProviders = Lookup.getDefault().lookupAll(CommonTestUtilProvider.class);
+	for (CommonTestUtilProvider provider : testUtilProviders) {
+	    locations = provider.getTestTargets(activeFO);
+	    break;
+	}
+	if (locations != null && locations.length == 0) {
+            SourceGroup sourceGroupOwner = findSourceGroupOwner(activeFO);
+            if (sourceGroupOwner != null) {
+                locations = UnitTestForSourceQuery.findUnitTests(sourceGroupOwner.getRootFolder());
+            }
+        }
+        return locations != null ? locations : new Object[0];
+    }
+
+    private static String getPreffiledName(FileObject fileObj, String selectedFramework) {
+	ClassPath cp = ClassPath.getClassPath(fileObj, ClassPath.SOURCE);
+	String className = cp.getResourceName(fileObj, '.', false);
+	return className + getTestingFrameworkSuffix(selectedFramework) + "Test";
+    }
+
+    private static String getTestingFrameworkSuffix(String selectedFramework) {
+	if (selectedFramework == null) {
+	    return "";
+	}
+	String testngFramework = "";
+	Collection<? extends GuiUtilsProvider> providers = Lookup.getDefault().lookupAll(GuiUtilsProvider.class);
+	for (GuiUtilsProvider provider : providers) {
+	    testngFramework = provider.getTestngFramework();
+	    break;
+	}
+	return selectedFramework.equals(testngFramework) ? "NG" : "";
+    }
+
+    private static FileObject getTargetFolder(Object selectedLocation) {
+	if (selectedLocation == null) {
+	    return null;
+	}
+	if (selectedLocation instanceof SourceGroup) {
+	    return ((SourceGroup) selectedLocation).getRootFolder();
+	}
+	assert selectedLocation instanceof FileObject;
```

Comment:
`getTargetFolderPath` accepts `URL` objects, this method does not. OK ?

Reply:
Fixed.